### PR TITLE
fix: delete associated works when bindings are deleted

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR fixes an issue where associated works are not deleted when the scheduler removes bindings.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### CODE REVIEW

**Comment by michaelawyu** [comment]
Closing this PR for now as the root cause seems to be elsewhere.

**Comment by michaelawyu** [comment]
This PR is sent as, when writing the E2E tests, it occurs that the system will not delete works derived from deleted bindings; logically it should be handled by the work generator (which creates the works). Though, when refreshing the tests it seems that the controller expects some kind of GC controller to do the work instead, which does not seem to exist at this moment. If the GC controller does exist, please let me know, and I would fix the issue differently.

File: pkg/controllers/workgenerator/controller.go [hunk 1]
```diff
@@ -160,6 +160,16 @@ func (r *Reconciler) handleDelete(ctx context.Context, resourceBinding *fleetv1b
 	if err != nil {
 		return ctrl.Result{}, err
 	}
+
+	// delete all the listed works
+	for workName := range works {
+		work := works[workName]
+
+		if err := r.Client.Delete(ctx, work); err != nil {
```
Comment (zhiying-lin) [inline pkg/controllers/workgenerator/controller.go:168]: ignore the notfound error?

File: pkg/controllers/workgenerator/controller_integration_test.go [hunk 2]
```diff
@@ -677,20 +677,30 @@ var _ = Describe("Test clusterSchedulingPolicySnapshot Controller", func() {
...
-					return k8sClient.Get(ctx, types.NamespacedName{Name: binding.Name}, binding)
-				}, duration, interval).Should(Succeed(), "the binding should not be removed until all the work is removed")
-				// delete the work (as the testEnv doesn't run GC controller)
-				Expect(k8sClient.Delete(ctx, &work)).Should(Succeed())
-				By(fmt.Sprintf("work  %s is deleted", work.Name))
-				// Check the binding is not still removed
-				Consistently(func() error {
-					return k8sClient.Get(ctx, types.NamespacedName{Name: binding.Name}, binding)
-				}, duration, interval).Should(Succeed(), "the binding should not be removed until all the work is removed")
-				// delete the work2 (as the testEnv doesn't run GC controller)
```
Comment (zhiying-lin) [inline pkg/controllers/workgenerator/controller_integration_test.go:691]: @ryanzhang-oss what's the GC controller? i cannot recall.

File: pkg/controllers/workgenerator/controller.go (hunk 1 above)
Comment (michaelawyu) [reply pkg/controllers/workgenerator/controller.go]: Done.

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**